server/src/main/java/org/opensearch/common/Rounding.java

```diff
@@ -696,6 +660,7 @@ public double roundingSize(long utcMillis, DateTimeUnit timeUnit) {
 
         private class FixedToMidnightRounding extends TimeUnitPreparedRounding {
             private final LocalTimeOffset offset;
+            private final JavaTimeToMidnightRounding MIDNIGHT_ROUNDING = new JavaTimeToMidnightRounding();
```

Comment:
Can we move `MIDNIGHT_ROUNDING` into `TimeUnitPreparedRounding`, so that it can be shared across various implementations of `TimeUnitPreparedRounding`